### PR TITLE
Document StateEvent for /createRoom

### LINKED ISSUE #1327: /createRoom has no descriptions for what a state event is
image

### PR DESCRIPTION
**Pull request by Half-Shot**
I mostly copied and pasted from https://matrix.org/docs/spec/client_server/r0.3.0.html#get-matrix-client-r0-rooms-roomid-state-eventtype-statekey but obviously changed it where needed. I also tidied up the English to read better e.g. "Defaults to the empty string" => "Defaults to an empty string"

Fixes #1327

### CODE REVIEW

File: api/client-server/create_room.yaml [hunk 1]
```diff
@@ -150,10 +150,13 @@ paths:
                   properties:
                     type:
                       type: string
+                      description: The type of event to send.
                     state_key:
                       type: string
+                      description: The state_key of the state event. Defaults to an empty string.
```
Comment (richvdh) [inline api/client-server/create_room.yaml]: is this not a requried attribute, and therefore has no default?

Comment (Half-Shot) [reply api/client-server/create_room.yaml]: I can't see why it would be? State events aren't required to have keys. I can stick an Optional on the front

Comment (richvdh) [reply api/client-server/create_room.yaml]: the very thing that makes it be a state event is that it has a (possibly empty) key.   Personally I would rather we were explicit about that.

worth checking what synapse currently requires imho

Comment (Half-Shot) [reply api/client-server/create_room.yaml]: I did :). Synapse defaults it to ""

Comment (Half-Shot) [reply api/client-server/create_room.yaml]:
```
        raw_initial_state = config.get("initial_state", [])

        initial_state = OrderedDict()
        for val in raw_initial_state:
            initial_state[(val["type"], val.get("state_key", ""))] = val["content"]
```

Comment (richvdh) [reply api/client-server/create_room.yaml]: UGH FINE THEN. still, type and content are required.

**Comment by Half-Shot** [comment]
I think that's all?

**Comment by richvdh** [comment]
thanks!

**Pull request merged**